code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -390,15 +394,24 @@
 		return 1
 
 /mob/living/simple_animal/death(gibbed)
-	health = 0
-	icon_state = icon_dead
-	stat = DEAD
-	density = 0
 	if(nest)
 		nest.spawned_mobs -= src
 		nest = null
-	if(!gibbed)
+	if(loot.len)
+		for(var/i in loot)
+			new i(loc)
+	if(deathmessage && !gibbed)
+		visible_message("<span class='danger'>[deathmessage]</span>")
+	else if(!del_on_death)
 		visible_message("<span class='danger'>\the [src] stops moving...</span>")
+	if(del_on_death)
+		ghostize()
+		qdel(src)
```

Comment:
Many creatures who normally delete on death don't do it if they have a ckey so that during Summon Magic, they can be hugboxed I mean ressed by bolts of ressurection.